### PR TITLE
Add Lobby, fix bug

### PR DESCRIPTION
**Pull request by BabichMikhail**

### CODE REVIEW

File: cwserver/NancyModules/LobbyModule.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                    return Response.AsRedirect("/games/view/" + game.Id.ToString());
+                } else if (Request.Form.JoinButton) {
+                    GameDatabase.AddUserToGame((UserModel)Context.CurrentUser, Request.Form.JoinButton);
+                    return Response.AsRedirect("/games/view/" + (string)Request.Form.JoinButton);
+                }
+                return Response.AsRedirect("/");
+            };
+
+            Get["/games/view/{id}"] = args => {
+                var game = GameDatabase.GetGame(args.id);
```
Comment (TheLocky) [inline cwserver/NancyModules/LobbyModule.cs]: It does not require authentication?

File: cwserver/NancyModules/LobbyModule.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                }
+                if (game.Status == GameModel.StatusActive) {
+                    return View["lobby/gameActive.cshtml", new { Game = game }];
+                } else { // game.status == Game.StatusWait
+                    var gameInfo = GameDatabase.GetGameInfo(args.id);
+                    return View["lobby/gameWait.cshtml", new { GameInfo = gameInfo, Game = game }];
+                }
+            };
+
+            Post["/games/view/{id}"] = args => {
```
Comment (TheLocky) [inline cwserver/NancyModules/LobbyModule.cs]: And it does not require authentication?

File: cwserver/NancyModules/LobbyModule.cs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                if (Request.Form.CreateButton) {
```
Comment (TheLocky) [inline cwserver/NancyModules/LobbyModule.cs]: Create = Start?

Comment (BabichMikhail) [inline cwserver/NancyModules/LobbyModule.cs]: No

**Comment by TheLocky** [comment]
DBConnection.cs conflict, fix it

File: cwserver/NancyModules/LobbyModule.cs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+using Nancy.Linker;
+using Nancy.Security;
+
+using cwserver.Models;
+
+namespace cwserver.NancyModules {
+    public class LobbyModule : NancyModule {
+        public LobbyModule(IResourceLinker linker) {
+            Get["/games"] = args => {
+                this.RequiresAuthentication();
```
Comment (TheLocky) [inline cwserver/NancyModules/LobbyModule.cs]: Maybe move this in constructor?

File: cwserver/Models/GameModel.cs [hunk 5]
```diff
@@ -0,0 +1,48 @@
+﻿namespace cwserver.Models {
+    public class GameModel {
+        public const int StatusActive = 1;
+        public const int StatusInactive = 2;
+        public const int StatusWait = 4;
```
Comment (TheLocky) [inline cwserver/Models/GameModel.cs]: Use enum

File: cwserver/Views/Lobby/games.cshtml [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+            <th>Join</th>
+            <th>Status</th>
+            <th>Info</th>
+        </thead>
+        <tbody>
+            @foreach (var game in Model.Games) {
+            <tr>
+                <td>@game.GameId</td>
+                <td>@game.Count</td>
+                <td><button type="submit" name="JoinButton" value="@game.GameId">join</button></td>
```
Comment (TheLocky) [inline cwserver/Views/Lobby/games.cshtml]: what if the user is already in another lobby?

File: cwserver/NancyModules/LobbyModule.cs [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+                return View["lobby/games.cshtml", new { Games = games }];
+            };
+
+            Post["/games"] = args => {
+                this.RequiresAuthentication();
+                if (Request.Form.CreateButton) { ;
+                    var game = GameDatabase.CreateGame(((UserModel)Context.CurrentUser).Id, GameModel.StatusWait);
+                    return Response.AsRedirect("/games/view/" + game.Id.ToString());
+                } else if (Request.Form.JoinButton) {
+                    GameDatabase.AddUserToGame((UserModel)Context.CurrentUser, Request.Form.JoinButton);
```
Comment (TheLocky) [inline cwserver/NancyModules/LobbyModule.cs]: what if the user is already in another lobby?

File: cwserver/Views/Lobby/games.cshtml (hunk 6 above)
Comment (BabichMikhail) [inline cwserver/Views/Lobby/games.cshtml]: yes, html comment. there is problem!

File: cwserver/Models/GameModel.cs (hunk 5 above)
Comment (BabichMikhail) [inline cwserver/Models/GameModel.cs]: enum is subset integer

Comment (BabichMikhail) [inline cwserver/Models/GameModel.cs]: and we don't know how many we will have statuses of games. Edit database for every new status?

**Pull request closed without merging**